src/ninemensmorris/BoardComp.java

```diff
@@ -0,0 +1,309 @@
+package nmmguisample;
+
+
+import com.almasb.checkers.CheckersApp;
+import java.util.HashMap;
+import javafx.scene.layout.Pane;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Circle;
+import javafx.scene.shape.Line;
+import javafx.scene.shape.Rectangle;
+import static nmmguisample.SaMPLENMM.HEIGHT;
+import static nmmguisample.SaMPLENMM.POS_SIZE;
+import static nmmguisample.SaMPLENMM.WIDTH;
+
+/*
+ *The board components ared designed using a blocked graph as referance.
+ */
+
+/**
+ * @author LENOVO
+ */
+public class BoardComp extends Rectangle{
+    private Coin coin;
+
+   
+    public boolean hasCoin(){
+        return coin != null;
+    }
+    public Coin getCoin(){
+        return coin;
+    }
+    public void setCoin(Coin coin){
+        this.coin = coin;
+    }
+    public BoardComp(boolean posMove, int x,int y ) {
+        setHeight(SaMPLENMM.POS_SIZE);
+        setWidth(SaMPLENMM.POS_SIZE);
+        relocate(x*SaMPLENMM.POS_SIZE,y*SaMPLENMM.POS_SIZE);
+        setFill(posMove ? Color.GREEN: Color.GREY);
+    }
+
+    public BoardComp() {
+    }
+    
+      public void PlaceCoin(Pane root,CoinType ct,double [] xy){
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.